public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts

```diff
@@ -0,0 +1,52 @@
+import { HistoryItem } from '@grafana/data';
```

Comment:
This is a data provider for completions that sits between the editor code and the `languageProvider` instance.